python/cuml/naive_bayes/naive_bayes.py

```diff
@@ -261,6 +261,383 @@ def predict_proba(self, X) -> CumlArray:
+            return ((X - X.mean(axis=axis)) ** 2).mean(axis=axis)
+
+        self.epsilon_ = self.var_smoothing * var_sparse(X).max()
+
+        if not self.fit_called_:
+            self.fit_called_ = True
+
+            # Original labels are stored on the instance
+            if _classes is not None:
+                check_labels(Y, _classes.to_output('cupy'))
+                self.classes_ = _classes
+            else:
+                self.classes_ = label_classes
+
+            n_features = X.shape[1]
+            n_classes = len(self.classes_)
+
+            self.n_classes_ = n_classes
+            self.n_features_ = n_features
+
+            self.theta_ = cp.zeros((n_classes, n_features))
+            self.sigma_ = cp.zeros((n_classes, n_features))
+
+            self.class_count_ = cp.zeros(n_classes, dtype=X.dtype)
+
+            if self.priors is not None:
+                if len(self.priors) != n_classes:
+                    raise ValueError("Number of priors must match number of"
+                                     " classes.")
+                if not cp.isclose(self.priors.sum(), 1):
+                    raise ValueError('The sum of the priors should be 1.')
+                if (self.priors < 0).any():
+                    raise ValueError('Priors must be non-negative.')
+                self.class_prior, *_ = input_to_cupy_array(
+                    self.priors,
+                    check_dtype=[cp.float32, cp.float64])
+
+        else:
+            self.sigma_[:, :] -= self.epsilon_
+
+        unique_y = cp.unique(y)
+        unique_y_in_classes = cp.in1d(unique_y, cp.array(self.classes_))
+
+        if not cp.all(unique_y_in_classes):
+            raise ValueError("The target label(s) %s in y do not exist "
+                             "in the initial classes %s" %
+                             (unique_y[~unique_y_in_classes], self.classes_))
+
+        self.theta_, self.sigma_ = self._update_mean_variance(X, Y)
+
+        self.sigma_[:, :] += self.epsilon_
+
+        if self.priors is None:
+            self.class_prior = self.class_count_ / self.class_count_.sum()
+
+        return self
+
+    def partial_fit(self, X, y, classes=None,
+                    sample_weight=None) -> "GaussianNB":
+        """
+        Incremental fit on a batch of samples.
+        This method is expected to be called several times consecutively on
+        different chunks of a dataset so as to implement out-of-core or online
+        learning.
+        This is especially useful when the whole dataset is too big to fit in
+        memory at once.
+        This method has some performance overhead hence it is better to call
+        partial_fit on chunks of data that are as large as possible (as long
+        as fitting in the memory budget) to hide the overhead.
+
+        Parameters
+        ----------
+
+        X : {array-like, cupy sparse matrix} of shape (n_samples, n_features)
```

Comment:
We might want to make a mention here that the "optimal" input for the sparse matrix is COOrdinate format, otherwise it will be copied in order to convert to that format.